From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 28 Sep 2019 08:11:49 +0200
Subject: [PATCH] add support for the Framed-MTU Radius attribute

http://ppp.samba.org/cgi-bin/ppp-bugs/incoming?id=1532

From: klepikov_a@up.ua
To: ppp-bugs@ppp.samba.org
Subject: Radius plugin does not set MTU on ppp interface
Date: Mon, 22 Jan 2007 12:36:59 +0000 (GMT)

Full_Name: Alexander Klepikov
Version: 2.4.3
OS: rhl 7.3 (2.4.20-28.7bigmem)
Submission from: (NULL) (213.130.21.73)


This patch allows radius plugin to deal with Framed-MTU Radius attribute and to
set MTU on interface.


Imported from ppp_2.4.7-2+4.1.debian.tar.xz

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 pppd/plugins/radius/radius.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pppd/plugins/radius/radius.c b/pppd/plugins/radius/radius.c
index 4ba5f523ea07..06e00590b635 100644
--- a/pppd/plugins/radius/radius.c
+++ b/pppd/plugins/radius/radius.c
@@ -651,6 +651,9 @@ radius_setparams(VALUE_PAIR *vp, char *msg, REQUEST_INFO *req_info,
 		    memcpy(rstate.class, vp->strvalue, rstate.class_len);
 		} /* else too big for our buffer - ignore it */
 		break;
+	    case PW_FRAMED_MTU:
+		netif_set_mtu(rstate.client_port,MIN(netif_get_mtu(rstate.client_port),vp->lvalue));
+		break;
 	    }
 
 
